bat-utils/lib/runtime-kafka.js

```diff
@@ -57,21 +57,39 @@ class Kafka {
       // parallel processing on topic level
       const topicPromises = Object.keys(batchOfMessages).map(async (topic) => {
         // parallel processing on partition level
-        const partitionPromises = Object.keys(batchOfMessages[topic]).map((partition) => {
+        let partition
+        let partKey = 0
+        while ((partition = batchOfMessages[topic][partKey])) {
           // sequential processing on message level (to respect ORDER)
-          const messages = batchOfMessages[topic][partition]
-
-          debug('batch', topic, messages.length, messages[0])
-          return this.topicHandlers[topic](messages)
-        })
-
-        // wait until all partitions of this topic are processed and commit its offset
-        // make sure to keep batch sizes large enough, you dont want to commit too often
-        await Promise.all(partitionPromises)
-        await consumer.commitLocalOffsetsForTopic(topic)
+          const partLength = partition.length
+          const offset = partition[partition.length - 1].offset
```

Comment:
I'm worried this isn't the right offset for us to be committing. [commitLocalOffsetsForTopic appears to work quite differently](https://github.com/nodefluent/node-sinek/blob/a90ba6c1b9d007c4d9577853500e68bba201c60c/lib/librdkafka/NConsumer.js#L894) - I would think we would want essentially the same logic it uses to determine the offset but only for a single topic / partition?

Reply:
Michael and I caught up about this - I got mixed up, here `partition` is the batch of messages from that partition - which has embedded offsets. so I believe this logic would work